lib/httpDigest.js

```diff
@@ -1,55 +1,100 @@
 /*!
  * Copyright (c) 2019-2020 Digital Bazaar, Inc. All rights reserved.
  */
-import base64url from './base64url.js';
-import base64 from './base64.js';
+import {encode as base64Encode} from 'universal-base64';
+import {encode as base64urlEncode} from 'base64url-universal';
 import crypto from './crypto.js';
 
 /**
- * Creates a string suitable for HTTP Digest: header.
+ * Creates a value suitable for the HTTP `Digest` header.
  *
- * @param {string|object} data - Objects to be hashed (typically, headers).
- * @param {string} algorithm - Hash algorithm to use. (e.g. 'sha256').
- * @param {boolean} useMultihash - Whether to encode via multihash.
+ * @param {object} options - The options to use.
+ * @param {string|object} options.data - Objects to be hashed
+ *   (typically headers).
+ * @param {string} [options.algorithm] - Hash algorithm to use.
+ *   (e.g. 'sha256').
+ * @param {boolean} options.useMultihash - Whether to encode via multihash.
  *   If false, the hash will be base64 encoded (non-url).
  *
- * @returns {Promise<string>} Resolves with hashed digest string.
+ * @returns {Promise<string>} Resolves to `Digest` header value.
  */
-async function create({data, algorithm = 'sha256', useMultihash = true}) {
+export async function createHeaderValue(
+  {data, algorithm = 'sha256', useMultihash = true} = {}) {
+  const {key, encodedDigest} = await _createHeaderValueComponents(
+    {data, algorithm, useMultihash});
+  return `${key}=${encodedDigest}`;
+}
+
+/**
+ * Verifies the HTTP `Digest` header value against the given HTTP body `data`.
+ *
+ * @param {object} options - The options to use.
+ * @param {string|object} options.data - The data to be verified.
+ * @param {string} options.headerValue - The digest header value to verify
+ *   the data against.
+ *
+ * @returns {object} - An object with `verified` set to `true` or `false`.
+ */
+export async function verifyHeaderValue({data, headerValue} = {}) {
+  const {key, algorithm, encodedDigest} = _parseHeaderValue(headerValue);
+  const {encodedDigest: expectedDigest} = await _createHeaderValueComponents(
+    {data, algorithm, useMultihash: key === 'mh'});
+  return {verified: encodedDigest === expectedDigest};
+}
```

Comment:
@dlongley do you agree with the following?

We have adopted the API contract for `verifyFoo` APIs that they do not throw.  They always return a true/false value along with error report as appropriate.

I recommend that all helpers being called here be wrapped in a try catch and...

The verify pattern used elsewhere is that all extraneous error will end up producing a return value something like:
```
{
  verified: false,
  error: e
}
```

Reply:
@mattcollier @dlongley I checked and it looks like that pattern is being used in other verifyFoos in other repos( for example [here](https://github.com/digitalbazaar/jsonld-signatures/blob/d81e605b910d5f5bcce52edac2334ff7b89212df/lib/suites/LinkedDataSignature2015.js#L80)). So I updated it - dad4dbf.